### PR TITLE
Bug 2087039: update dependencies to K8s 1.24, go 1.18

### PR DESCRIPTION
**Pull request by damdo**
- Update the dependencies to K8s 1.24.
- Bump project to go 1.18.

### CODE REVIEW

**Review by alexander-demichev — APPROVED** [review]
/approve

File: hack/fetch-ext-bins.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ if [ -n "$TRACE" ]; then
   set -x
 fi
 
-k8s_version=1.22.1
+k8s_version=1.23.5
```
Comment (lobziik) [inline hack/fetch-ext-bins.sh:29]: should it be 1.24.something?

Comment (lobziik) [reply hack/fetch-ext-bins.sh:29]: ok, no kubebuilder for 1.24 yet, ignore this

Comment (damdo) [reply hack/fetch-ext-bins.sh:29]: We always lag behind several versions with those envtest kubebuilder tools (On some repos we where even on 1.19). As they release them significantly later unfortunately. That's why we were still on 1.22 on the previous release. For now we are stuck on 1.23.x.

**Pull request merged**